### PR TITLE
feature: use rechoir

### PR DESCRIPTION
**Pull request by nfischer**
This is a redo of #258, since it looks like it may have been abandoned.

### CODE REVIEW

**Comment by nfischer** [comment]
@TimothyGu thanks for the command. Fixed it.

If someone could please review this, that would be much appreciated.

File: test/shjs.js [hunk 1]
```diff
@@ -4,10 +4,9 @@ var assert = require('assert');
 
 function runScript(name) {
   // prefix with 'node ' for Windows, don't prefix for OSX/Linux
-  return shell.exec((process.platform === 'win32' ? 'node ' : '') +
-                    path.resolve(__dirname, '../bin/shjs') +
-                    ' ' +
-                    path.resolve(__dirname, 'resources', 'shjs', name), { silent: true });
+  var cmd = (process.platform === 'win32' ? 'node ' : '') + path.resolve(__dirname, '../bin/shjs');
```
Comment (TimothyGu) [inline test/shjs.js]: I'm not sure if I have already mentioned this, but the best way to spawn another `node` process is through `process.argv[0]`. On both Unix and Windows, `node` refers to the executable in PATH, while the user might have executed the script with the absolute path to another `node` executable. `process.argv[0]` guarantees that the current used Node.js is always used.

Comment (nfischer) [inline test/shjs.js]: Yeah, that works. So does `process.execPath` I believe. I'll adjust.

**Comment by nfischer** [comment]
Updated the tests to use `process.execPath` instead of "`node`" to make things more robust.

**Comment by ariporad** [comment]
It looks like the tests aren't passing, otherwise looks good.

**Comment by nfischer** [comment]
Yeah, I see the issue. I just need to wrap process.execPath with JSON.stringify(). Or you can take out the last commit and merge that (it's only the execPath stuff that broke tests). Away from my computer at the moment

**Comment by ariporad** [comment]
LGTM, will merge once tests pass.

**Comment by nfischer** [comment]
Sorry, just found another issue. Now it should be resolved.

**Comment by nfischer** [comment]
Merging this, since it LGTM

**Pull request merged**